tools/optimize/file_optimizer.cpp

```diff
@@ -102,6 +102,22 @@ bool FileOptimizer::FilterInitBufferMetaData(const format::BlockHeader& block_he
         // If the buffer is in the unused list, omit its initialization data from the file.
         if (unreferenced_ids_.find(header.buffer_id) != unreferenced_ids_.end())
         {
+            // In its place insert a dummy annotation meta command. This should keep the block index when
+            // replaying an optimized trimmed capture in in alignment with the block index calculated
+            // at capture time
+            format::AnnotationHeader annotation;
+            annotation.block_header.size = format::GetAnnotationBlockBaseSize();
+            annotation.block_header.type = format::BlockType::kAnnotation;
+            annotation.annotation_type   = format::kUnknown;
```

Comment:
Could this annotation contain text instead or even JSON to the effect that it replaced an omitted block?  I think it would be helpful if the annotation even said what was omitted.

Reply:
sure